packages/release/activate-configured.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
```

Comment:
I'm a bit confused by the use of procedural units to provide what I think are intended to be "hard" synchronization points. systemd provides the means to express the same graph without intermediate `activate-<next>.service` units by way of `<next>.target`s and job ordering. 

To me, the use of `systemctl isolate` seems error prone as unexpressed dependencies may result in a half-working system where a modification of transitive dependency could actually drop a unit from the final set of running units.

This change is somewhat new to me so I might be missing some context - so please forgive me if this was already discussed:

* Did we determine that a `configured-pre.target` and `configured.target` would be unable to express the desired ordering?

* Does `systemctl isolate` provide desired functionality that unit ordering and dependency expression does not? What do we gain by injecting a corrective set of jobs (with `systemctl isolate`) at runtime vs. using dependency ordering?

* Can we avoid replacing the default unit and instead order our units within the default graph?

Reply:
> Did we determine that a `configured-pre.target` and `configured.target` would be unable to express the desired ordering?

Yes, since the problem that we have is that systemd doesn't start runtime created units (like host containers) unless we force them to start (a.k.a `systemctl enable <> --now`). Force starting units results in units starting/stopping multiple times if any of their dependencies fails to start.

>Does `systemctl isolate` provide desired functionality that unit ordering and dependency expression does not? What do we gain by injecting a corrective set of jobs (with `systemctl isolate`) at runtime vs. using dependency ordering?

Yes, since we can dynamically update the jobs graph at runtime which allows us to add more dependencies to the targets if needed (a.k.a. bootstrap containers). And we gain the ability to dynamically "update" the dependency graph and load the units at the correct time, without forcing starts with `systemctl enable --now`

> Can we avoid replacing the default unit and instead order our units within the default graph?

No, we need `preconfigured` to be the default target for the boot, and transition along the targets to load the systemd services (potentially created during `preconfigured`) without force start them (like we used to do with `host-contaieners`).

Reply:
Thanks for writing back here. The details will be helpful to maybe-future us scratching our heads reading this. 

We discussed this a bit more offline - this all is roughly what I suspected. I think we should stick with the solution you have implemented here. If we run into issues later on then we can revisit it.

I'm not a fan of `systemctl isolate` by any means, but really this incidental complexity is `systemd`'s fault 😆